### ISSUE TITLE
[all packages with scheduled tasks] -add package parameter to also execute in battery mode

### ISSUE THREAD

**Issue by DJCrashdummy** [bcurran3/ChocolateyPackages_CCR#69]
all tasks (no matter how small they are) of your packages are set to *"**not** execute while the machine is in battery mode"*.
may i guess this is unintended resp. by accident because this is the default or something...?

IMHO this should be deactivated and also the *"abortion if the computer changes to battery mode"*.

or another option would be to a least activate *"execute the task as soon as possible after a missed appointment"* (if this works for battery-/ac-mode), but im not sure...
i noticed this just by accident because a machine didn't get updates for months although it was definitively running at the scheduled times... but obviously just in battery mode.

**Comment by bcurran3** [comment]
ERRATA: No tasks are set that way, Microsoft defaults tasks that way.

There are reasons for not doing things while running via battery and why tasks default that way. I'm not sure I want to bump up against those reasons.

Mobile devices are problematic for me as well. Between sporadic power on times, not wanting to bother users during working hours, and tasks not running on batteries, it's hard to keep them up to date. Currently I tend to set choco-upgrade-all-at at 12:30pm hoping the computer is powered on and not being used during lunchtime, but that's not a perfect solution either.

I'm inclined to say I'm not going to fix this as is, but it's another one where a new package parameter isn't hard to do to let let you choose this default behavior change so I it goes on the TDL.

I am very inclined to add a retry for missed scheduled events ala https://github.com/bcurran3/ChocolateyPackages/issues/48.

**Comment by DJCrashdummy** [comment]
> ERRATA: No tasks are set that way, Microsoft defaults tasks that way.

>>  may i guess this is **unintended** resp. by accident **because this is the default** or something...?


for sure there are reasons to not execute some tasks while in battery mode, but let me show you a different approach:
people are installing your packages for a reason to get things (automatically) done... and it is even more annoying if this tasks are still not executed although they expect it otherwise - especially regarding security, which e.g. backups and updates clearly are.
if some people are concerned about their battery status they either don't install your package for unattended execution anyway or dig deeper if exclusions are possible.
in short: i set up a automatic tasks because i want things to be done, and then (if necessary) care about exceptions or even not...

please do not make the exceptions the default (even if microsoft obviously does this)... but just because microsoft does something, it doesn't have to be the best or even make sense! :slightly_frowning_face: 

so may i suggest to execute all tasks also in battery mode by default and either
- at least mention in the description that this can be changed after installing in the tasks,
- create an optional parameter to **not** execute in battery mode
- or completely ignore the exception, because people install your packages for a reason.

**Comment by DJCrashdummy** [comment]
additionally: i would see this topic as independent as possible from #48 (resp. at least for the default setting), because often people (and me too) plug the notebook quickly in while working because i have to, or before/while presentations to not suffer an "accident"... and having in this situations executed a bunch of tasks (because they are by default not executed on time in battery mode, but then made up at "some random time") is even more annoying!

**Comment by DJCrashdummy** [comment]
to sum it up (not just for this issue): i would advise in general to make all tasks...
- free to chose the execution time (by package parameter)
- execute in battery mode by default
- add an optional parameter to not execute in battery mode
- add an optional parameter to execute as soon as possible after a scheduled start is missed

and then making "execution as soon as possible after a scheduled start is missed" the default for short tasks (as suggested in #48) can be considered, if the divergence makes sense. - for sure then the optional package parameter would have to be to deactivate this behaviour.

**Comment by bcurran3** [comment]
Was about to batch add the battery support to the scheduled tasks as I THOUGHT it was going to be fairly simple and fast, BUT found out that schtasks.exe doesn't support the feature. I have to re-write the task creation code using powershell functions (which I have to re-review) or have schtasks create the tasks via loading an XML to use the <DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries> option. Either way it's not as simple as I hoped and will have to stave off for yet another day. :(
REF: https://github.com/chef/chef/issues/6403

**Comment by bcurran3** [comment]
Closing this as I am no longer creating or maintaining Chocolatey packages. Sorry, out
of my control.

Hopefully you've enjoyed my work and found it useful.

[![Bella Ciao!](https://img.youtube.com/vi/9ao4FEaDGhQ/0.jpg)](https://www.youtube.com/watch?v=9ao4FEaDGhQ)
